sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -417,6 +417,57 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
           Context const& context) const override;
     };
 
+    /**
+     * @brief Helper class for creating scopes for the token request context.
+     *
+     */
```

Comment:
Is this policy.hpp header the right place for such a class and helper function?

Reply:
It is there so it becomes available right now for anyone using the `BearerTokenAuthenticationPolicy`, as that's the policy which requires the scopes

Reply:
It is closely related to `Azure::Core::Credentials::TokenRequestContext`.
Move it there (to `_internal` namespace), name it `TokenRequestContextHelper`, and name the method `GetScopeFromServiceUrl()`